### PR TITLE
handling multiple build environments

### PR DESCRIPTION
**Pull request by aalu1418**
Keep using a single set of program IDs in the program files
* Use scripts to modify and build for local testing and staging environment

### CODE REVIEW

File: scripts/lib.sh [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env bash
```
Comment (archseer) [inline scripts/lib.sh:1]: Use `set -euxo pipefail` on all bash scripts: https://vaneyckt.io/posts/safer_bash_scripts_with_set_euxo_pipefail/

File: scripts/setup-staging.sh [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#!/usr/bin/env bash
+
+# note call from repo root
```
Comment (archseer) [inline scripts/setup-staging.sh:3]: You can make it work relative to the script by using `${BASH_SOURCE%/*}` http://mywiki.wooledge.org/BashFAQ/028

**Comment by aalu1418** [comment]
i'm pretty sure the `e2e_tests_custom_cl_image / Build custom CL image / Build Chainlink Image (push)` test in CI will fail because there were some breaking changes introduced in core related to creating bootstrap jobs. (long story short, bootstrap jobs are created with the `bootstrap` job type now rather than a flag in a spec)

**Comment by archseer** [comment]
You need a rebase, @skudasov fixed this in https://github.com/smartcontractkit/chainlink-solana/pull/226

**Review by archseer — APPROVED** [review]

**Pull request merged**